src/Microsoft.AspNetCore.Authentication.OAuth/OAuthHandler.cs

```diff
@@ -223,10 +223,14 @@ protected virtual string BuildChallengeUrl(AuthenticationProperties properties,
             return QueryHelpers.AddQueryString(Options.AuthorizationEndpoint, parameters);
         }
 
+        [Obsolete("Subclasses should override FormatScope(IEnumerable<string>) instead")]
```

Comment:
We're going to have to be more clever about this, it breaks existing providers like Reddit and they're forced to update. The challenge properties support needs to be more opt-in.
How about only calling the new method if there's an override? And falling back to the existing method otherwise?

Reply:
Updated the PR in a way that hopefully fixes this issue. I’m using the `FormatScope` now whenever we want to format the `Options.Scope` value, and only call the new overload when we use an overwritten scope collection. That way, existing handlers that had `FormatScope()` overridden should still work fine—except when users attempt to overwrite the scopes using the AuthenticationProperties parameters. Then the providers will have to act and provide an overload for the `FormatScopes(IEnumerable<string>)` too. But I guess we cannot avoid that.